addons/controllers/cpi/cpiconfig_controller.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package controllers implements k8s controller functionality for vsphere-cpi.
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cpiv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cpi/v1alpha1"
+)
+
+// CPIConfigReconciler reconciles a CPIConfig object
+type CPIConfigReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=cpi.tanzu.vmware.com,resources=cpiconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cpi.tanzu.vmware.com,resources=cpiconfigs/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the CPIConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
```

Comment:
suggestion: lets remove those boilerplate code?

Reply:
yes, I will remove this boilerplate once I finish the controller login in another separate PR.

Reply:
trying to understand a little more on what is blocking us from having above cleaned up within this PR?

Reply:
removed the boilerplate. adds a TODO with issue link.